contracts/6/interfaces/IERC20Permit.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.6.9;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IERC20Detailed is IERC20 {
+  function name() external view returns(string memory);
+  function decimals() external view returns(uint256);
+}
+
+interface IERC20Nonces is IERC20Detailed {
+  function nonces(address holder) external view returns(uint);
+}
+
+interface IERC20Permit is IERC20Nonces {
+  function permit(address holder, address spender, uint256 nonce, uint256 expiry,
+                  bool allowed, uint8 v, bytes32 r, bytes32 s) external;
+
+  function permit(address holder, address spender, uint256 value, uint256 expiry,
```

Comment:
we should give different names here to avoid confusion. 
like permit and permitEIP2612

Reply:
I see. ok doing it

Reply:
thats not possible cause token contract would still have function selector permit so we can't change this in interface.